.github/workflows/publish_cli.yml

```diff
@@ -19,13 +19,30 @@ name: publish_cli
 on:
   push:
     tags:
-      - 'iggy-cli-[0-9]+.[0-9]+.[0-9]+'
+      - 'iggy-cli-*'
 
 env:
   GITHUB_TOKEN: ${{ github.token }}
   CARGO_TERM_COLOR: always
 
 jobs:
+  validate:
+    if: startsWith(github.ref, 'refs/tags/iggy-cli-')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Extract tag name
+        id: extract
+        run: echo "TAG=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+
+      - name: Validate tag format
+        run: |
+          TAG=${TAG}
+          if [[ ! "$TAG" =~ ^iggy-cli-([0-9]{1,3})\.([0-9]{1,3})\.([0-9]{1,3})$ ]]; then
+            echo "Tag $TAG does not match strict semver format (iggy-cli-X.Y.Z where 0 <= X,Y,Z <= 999)"
+            exit 1
+          fi
+          echo "Valid tag: $TAG"
+
   tag:
```

Comment:
I think it would be good to add restriction that `tag` step needs `validate` step to be completed - otherwise `validate` may fail while `tag` will run in parallel successfully creating release from invalid git tag. 

Same comment for remaining 2 workflows.